### PR TITLE
[R-package] Update lgb.model.dt.tree.R to use keyword arguments

### PR DESCRIPTION
**Pull request by zenggyu**
Addresses part of #3390. I have regenerated the documents, but `git diff` shows no differences in `.Rd` files.

### CODE REVIEW

File: R-package/R/lgb.model.dt.tree.R [hunk 1]
```diff
@@ -111,7 +111,10 @@ single.tree.parse <- function(lgb_tree) {
         , leaf_count = integer(0L)
       )
       # start tree traversal
-      pre_order_traversal(env, tree_node_leaf, current_depth, parent_index)
+      pre_order_traversal(env = env
+                          , tree_node_leaf = tree_node_leaf
+                          , current_depth = current_depth
+                          , parent_index = parent_index)
```
Comment (jameslamb) [inline R-package/R/lgb.model.dt.tree.R:117]:
```suggestion
      pre_order_traversal(
          env = env
          , tree_node_leaf = tree_node_leaf
          , current_depth = current_depth
          , parent_index = parent_index
      )
```

Can you please use this indentation to match the rest of the package?

**Review by jameslamb — CHANGES_REQUESTED** [review]
Thanks very much for taking the time to contribute to `{lightgbm}`! This looks great, just one small style change request.

Don't worry about re-generating the documentation again when you change it. I just put that in the description of #3390 because some PRs for that issue might required it and some might not. Looks like this one did not.

File: R-package/R/lgb.model.dt.tree.R (hunk 1 above)
Comment (zenggyu) [reply R-package/R/lgb.model.dt.tree.R]: OK!

File: R-package/R/lgb.model.dt.tree.R [hunk 2]
```diff
@@ -111,7 +111,12 @@ single.tree.parse <- function(lgb_tree) {
         , leaf_count = integer(0L)
       )
       # start tree traversal
-      pre_order_traversal(env, tree_node_leaf, current_depth, parent_index)
+      pre_order_traversal(
+        env = env
+        , tree_node_leaf = tree_node_leaf
+        , current_depth = current_depth
+        , parent_index = parent_index
+        )
```
Comment (jameslamb) [inline R-package/R/lgb.model.dt.tree.R:119]: can you please align this `)` with the `p` in `pre`?

```suggestion
      )
```

**Review by jameslamb — CHANGES_REQUESTED** [review]

File: R-package/R/lgb.model.dt.tree.R (hunk 1 above)
Comment (jameslamb) [reply R-package/R/lgb.model.dt.tree.R]: thanks! I saw you pushed a commit...in the future, you might find it easier to click the "Commit suggestion" button in the UI

image

Comment (zenggyu) [reply R-package/R/lgb.model.dt.tree.R]: Oh, OK, didn't know about that!

**Review by jameslamb — APPROVED** [review]
Looks good @zenggyu , thanks for taking the time to contribute to LightGBM! Come back any time!

**Pull request merged**